ikarus/finiteelements/febases.hh

```diff
@@ -0,0 +1,180 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+/**
+ * \file febases.hh
+ * \brief Contains the FEBases class, which works with different types of bases in FlatInterLeaved elements.
+ */
+
+#pragma once
+
+#include <ikarus/finiteelements/fetraits.hh>
+
+namespace Ikarus {
+
+/**
+ * \brief FEBases class for working with different types of bases in FlatInterLeaved elements.
+ *
+ * \details A single class which can be used by the finite elements to handle scalar basis, power basis, and composite
+ * basis. This class assumes the local indices in a FlatInterLeaved order.
+ *
+ * \tparam B The type of the basis.
+ */
+template <typename B>
+class FEBases
```

Comment:
This class should handle in the end the flat global indices for all our finite elements. Thus, I think it should be better named: `FlatGlobalIndices`

This will be inline with my future refactoring of an fe skill system: https://godbolt.org/z/jjaEYMYY3

It will work like `makeBasis(gridView,preBasis)` such that we can write: `makeFE(linearElastic(5),skills(neumannBoundaryCondition(19),volumeLoad()))` 

then in the end this class we have here is a very fundamental "skill", which communicates the global indices: therfore i would propose the following name `FlatGlobalIndices`.

Reply:
As we discussed, I moved `globalFlatIndices` function as a free function in `FEHelper` namespace. Then what do you think of the name `BasisSkills` for this class as it helps to provide all the underlying functionalities of a basis to a finite element?

Reply:
I went with `FEBase` as the class name here as it is the base class from which all the finite elements must inherit from. 
I am not sure if it creates any confusion with the term `basis` here. 
The class itself gets the `globalBasis` and thereby binds the `localView` to the `element` and helps to provide some of the information related to the local view. May be we can also name the class as `LocalViewFE`? 
Since it denotes a base class to all the finite elements, may be we can also call it `FiniteElement`, from which all finite elements are derived, but may be then its purpose is not clear?
What do you think? @rath3t